openedx/adg/lms/applications/api_views.py

```diff
@@ -109,10 +115,24 @@ class WorkExperienceViewSet(ApplicationRequirementsViewSetBaseClass):
 
     partial_update:
         Update an existing work experience record.
+        Update user work_experience is not applicable
 
     **Example Requests**
     PATCH /api/applications/work_experience/
+    PATCH /api/applications/work_experience/update_is_not_applicable/
 
     """
     queryset = WorkExperience.objects.all()
     serializer_class = WorkExperienceSerializer
+
+    @action(detail=False, methods=['patch'],  url_name='update_is_not_applicable', url_path='update_is_not_applicable')
+    def update_is_work_experience_not_applicable(self, request):
+        is_work_experience_not_applicable = json.loads(request.data.get('is_work_experience_not_applicable', 'false'))
+        user_application = request.user.application
+        user_application.is_work_experience_not_applicable = is_work_experience_not_applicable
+        with transaction.atomic():
+            user_application.save()
+            if is_work_experience_not_applicable:
+                WorkExperience.objects.filter(user_application=user_application).delete()
```

Comment:
We can change it to `user.application.applications_workexperiences.all().delete()` more django specific.